internal/consul/version_test.go

```diff
@@ -187,3 +187,63 @@ func Test_VersionsToString(t *testing.T) {
 		})
 	}
 }
+
+func Test_GetLatestPatch(t *testing.T) {
+	tcs := map[string]struct {
+		input    string
+		expected string
+		found    bool
+	}{
+		"Invalid": {
+			input:    "invalid",
+			expected: "",
+			found:    false,
+		},
+		"NotFound": {
+			input:    "1.1.0",
+			expected: "",
+			found:    false,
+		},
+		"Found": {
+			input:    "1.13.1",
+			expected: "1.13.3",
+			found:    true,
+		},
+		"FoundAlreadyLatest": {
+			input:    "1.13.3",
+			expected: "1.13.3",
+			found:    true,
+		},
+	}
+
+	for n, tc := range tcs {
+		t.Run(n, func(t *testing.T) {
+			r := require.New(t)
+
+			versions := []*consulmodels.HashicorpCloudConsul20210204Version{
+				{
+					Version: "v1.13.2",
+				},
+				{
+					Version: "v1.12.5",
+				},
+				{
+					Version: "v1.11.10",
+				},
+				{
+					Version: "v1.14.0",
+				},
+				{
+					Version: "v1.13.3",
```

Comment:
I think the order of these tests is going to guarantee a pass when you could get 1.13.1 after this and it would return that version.